src/Data/Vec/Base.agda

```diff
@@ -280,14 +286,30 @@ fromList (List._∷_ x xs) = x ∷ fromList xs
 ------------------------------------------------------------------------
 -- Operations for reversing vectors
 
-reverse : ∀ {n} → Vec A n → Vec A n
-reverse {A = A} = foldl (Vec A) (λ rev x → x ∷ rev) []
+-- snoc
 
 infixl 5 _∷ʳ_
 
-_∷ʳ_ : ∀ {n} → Vec A n → A → Vec A (1 + n)
+_∷ʳ_ : ∀ {n} → Vec A n → A → Vec A (suc n)
 []       ∷ʳ y = [ y ]
 (x ∷ xs) ∷ʳ y = x ∷ (xs ∷ʳ y)
+{- better ?
+_∷ʳ_ {A = A} xs y = foldr ((Vec A) ∘ suc) (λ x r → x ∷ r) [ y ] xs
```

Comment:
But see my comment above. BTW, `_∷ʳ_` was already defined; it was the `foldl`-based definition of `_ʳ++_`  which had me reconsidering those functions defined as `foldr`s... towards some grand (folly of a) general `foldl`/`foldr` commutation result.

Reply:
So here's a strange thing: I was doing some more reorganising, to try to see if `unfold-reverse` and `unfold-ʳ++` could be rewritten in terms of `foldr-reverse` (you'd think, right? at least in view of the above ;-)) and find myself in a situation where agda  seems reluctant to unfold the recursion equations for `foldl` correctly, even with every argument fully written out... starting to wonder if there is a bug?

Reply:
Or then again, it perhaps would *require* `_∷ʳ_` to be written as a `foldr` for that idea to work... instead of being, clause for clause, identical to a `foldr` instance... hmmm.

Reply:
Above conjecture correct. So... functions defined by pattern-matching are not intensionally equal to (suitable instances of slightly more higher-order) functions with the same pattern-matching behaviour. Hmmm. Generativity bites again? Oh well.

Reply:
And... `List.Properties` again leads the way with a bunch of `*-is-foldr` lemmas. I'll leave those to another day perhaps